lib/winston-logstash.js

```diff
@@ -166,7 +166,7 @@ Logstash.prototype.connect = function () {
 
     self.socket = null;
 
-    if (!ECONNREFUSED_REGEXP.test(err.message)) {
+    if (err.message.indexOf('E') !== 0) {
```

Comment:
Being honest we did not find the real problem yet. We have a very tricky Internet - and we keep having `winston-logstash` crashing our whole stack when Internet is down, even after setting `max_connect_retries: -1`.